app/scripts/app-init.js

```diff
@@ -111,14 +99,13 @@ function importAllScripts() {
   }
 }
 
-// eslint-disable-next-line no-undef
-self.addEventListener('install', importAllScripts);
+importAllScripts();
```

Comment:
+1 to it being easier to reason about.

We should see if there's a measurable difference in loading behavior (via performance/JS profiler/tracing) between importing top-level vs importing on install.

Reply:
Actually service worker documents also suggest to use `install` event for loading assets. I have doubt if this approach is the best.